### ISSUE TITLE
Feature Request: Allow host names in addition to ip-addresses

### ISSUE THREAD

**Issue by Jeroen-R** [RiptideNetworking/Riptide#147]
When specifying the server to connect to, it would be useful to allow hostnames. 

If I take this library into production there will be multiple servers, which makes management of those servers an important topic.

When I run the following code, I'd expect it to work, but I get an error messages

            `Client.Connect("localhost:7777");`

> (CLIENT): Invalid host address 'localhost:58473'! IP and port should be separated by a colon, for example: '127.0.0.1:7777'.

This appears to be a bit of an artificial limitation, I have never run into a system where I cannot swap out the ip address for the host name (I am sure it happens, just seen it rarely)

I am on 2.1.2.

**Comment by tom-weiland** [comment]
I wouldn't necessarily call it an "artificial" limitation, it's just not being handled automatically by Riptide—any time a system you use connects to a host name, that is really just being resolved to an IP address first and that's what you're *actually* connecting to.

You can use the .NET [`Dns.GetHostEntry`](https://learn.microsoft.com/en-us/dotnet/api/system.net.dns.gethostentry) method to retrieve the IP address(es) from a host name. Then you can pick the one you want (it can return multiple addresses since DNS supports more than one record) and pass it to Riptide to connect to it.

I considered adding a helper method for this to Riptide, but never got around to it before I stepped away from the project. I'm also not sure how helpful that would be considering people may have vastly different requirements regarding which of the returned IP addresses gets used (when there's more than one).

**Comment by cyraid** [comment]
@tom-weiland are there any other maintainers/developers? If you're the only one, could you at least write a message in the readme that it's abandoned or looking for maintainers etc?

**Comment by tom-weiland** [comment]
> are there any other maintainers/developers?

@xzippyzachx has the perms necessary to commit and merge PRs, and he has implemented a few bug fixes since I stopped working on it. Overall Riptide is in a pretty good place though and even if I was still working on it there wasn't much else I wanted to add, so I'm not quite sure "abandoned" is the right way to describe it.

**Comment by cyraid** [comment]
@tom-weiland fair enough. I suppose it's more the qualify of life improvements like this issue and kinda expecting being able to have the library call the DNS lookup for you, etc. Would the others be in control of adding such changes?

**Comment by tom-weiland** [comment]
> Would the others be in control of adding such changes?

Anyone can fork the repo and submit a pull request, and as I mentioned @xzippyzachx has the necessary perms to merge PRs.
